pkg/alert/client_test.go

```diff
@@ -0,0 +1,183 @@
+package alert
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/thanos-io/thanos/pkg/discovery/dns"
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+func TestBuildAlertmanagerConfiguration(t *testing.T) {
+	for _, tc := range []struct {
+		address string
+
+		err      bool
+		expected AlertmanagerConfig
+	}{
+		{
+			address: "http://localhost:9093",
+			expected: AlertmanagerConfig{
+				StaticAddresses: []string{"localhost:9093"},
+				Scheme:          "http",
+			},
+		},
+		{
+			address: "https://am.example.com",
```

Comment:
Currently, we did not support HTTPS, so that's nice as well :+1:  worth to mention in CHANGELOG I think, similar to comment below

Reply:
You could already pass `--alertmanagers.url https://foo.example.com`?